safe/impact_function/test/test_multi_exposure_impact_function.py

```diff
@@ -0,0 +1,89 @@
+    MultiExposureImpactFunction)
+
+
+LOGGER = logging.getLogger('InaSAFE')
+
+__copyright__ = "Copyright 2017, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+qgis_iface()
+
+
+class TestImpactFunction(unittest.TestCase):
+
+    """Test Multi Exposure Impact Function."""
+
+    def test_bad_multi(self):
+        """Test that the 'prepare' state can failed."""
+        hazard_layer = load_test_vector_layer(
+            'gisv4', 'hazard', 'classified_vector.geojson')
+        population_layer = load_test_vector_layer(
+            'gisv4', 'exposure', 'population.geojson')
+        population_bis = load_test_vector_layer(
+            'gisv4', 'exposure', 'population_multi_fields.geojson')
+        roads_layer = load_test_vector_layer(
+            'gisv4', 'exposure', 'roads.geojson')
+        aggregation_layer = load_test_vector_layer(
+            'gisv4', 'aggregation', 'small_grid.geojson')
+
+        # Same population exposure
+        impact_function = MultiExposureImpactFunction()
+        impact_function.hazard = hazard_layer
+        impact_function.exposures = [population_layer, population_bis]
+        impact_function.aggregation = aggregation_layer
+        self.assertTrue(
+            impact_function.prepare()[0] == PREPARE_MULTI_FAILED_EXPOSURES)
+
+        # Missing exposure layer
+        impact_function = MultiExposureImpactFunction()
+        impact_function.hazard = hazard_layer
+        impact_function.exposures = []
+        impact_function.aggregation = aggregation_layer
+        self.assertTrue(
+            impact_function.prepare()[0] == PREPARE_MULTI_FAILED_EXPOSURES)
+
+        # Normal multi exposure
+        impact_function = MultiExposureImpactFunction()
+        impact_function.hazard = hazard_layer
+        impact_function.exposures = [population_layer, roads_layer]
+        impact_function.aggregation = aggregation_layer
+        self.assertTrue(
+            impact_function.prepare()[0] == PREPARE_MULTI_SUCCESS)
+
+    def test_multi_exposure(self):
+        """Test we can run a multi exposure analysis."""
+        hazard_layer = load_test_vector_layer(
+            'gisv4', 'hazard', 'classified_vector.geojson')
+        building_layer = load_test_vector_layer(
+            'gisv4', 'exposure', 'building-points.geojson')
+        population_layer = load_test_vector_layer(
+            'gisv4', 'exposure', 'population.geojson')
+        roads_layer = load_test_vector_layer(
+            'gisv4', 'exposure', 'roads.geojson')
+        aggregation_layer = load_test_vector_layer(
+            'gisv4', 'aggregation', 'small_grid.geojson')
+
+        impact_function = MultiExposureImpactFunction()
+        impact_function.hazard = hazard_layer
+        impact_function.exposures = [
+            building_layer, population_layer, roads_layer]
+        impact_function.aggregation = aggregation_layer
+
+
+
+
```

Comment:
__[flake8]__
* [W391] blank line at end of file

<!-- Comment by [SideCI](https://sideci.com) -->